components/browser/session/src/main/java/mozilla/components/browser/session/LegacySessionManager.kt

```diff
@@ -398,20 +399,44 @@ class LegacySessionManager(
         return findNearbySession(index) { !it.isCustomTabSession() }
     }
 
+    @Suppress("LongMethod") // Yes, this method is pretty long. I hope we can delete it soon.
     private fun findNearbySession(index: Int, predicate: (Session) -> Boolean): Int {
-        val maxSteps = max(values.lastIndex - index, index)
+        // Okay, this is a bit stupid and complex. This implementation intends to implement the same behavior we use in
+        // BrowserStore - which is operating on a list without custom tabs. Since LegacySessionManager uses a list with
+        // custom tabs mixed in, we need to shift the index around a bit.
+
+        // First we need to determine the number of custom tabs that are *before* the index we removed from. This is
+        // later needed to calculate the index we would have removed from in a list without custom tabs.
+        var numberOfCustomTabsBeforeRemovedIndex = 0
```

Comment:
nit: could write as `sublist.filter.size` but not sure that's much easier to read.

Reply:
Tried `sublist.filter.size` and I like it! Also with that this can be `val` instead of `var`.

It doesn't like an invalid range though (e.g. `fromIndex(0) > toIndex(-1)`)

However with `take(max(0, ....))` that works. It's not easier to read though. :D

Reply:
Oh, mh, now I have a test failure. I'll change it back. Don't want to look at that code any longer... :)